### PR TITLE
Support LLM assisted tagging

### PR DESCRIPTION
**Pull request by AdityaKhatri**
## Changes

* Support LLM assisted tagging
* Remove assisted tagging mapping interface

## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] build works
- [ ] eslint issues
- [ ] typescript issues
- [ ] codegen errors
- [ ] `console.log` meant for debugging
- [ ] typos
- [ ] unwanted comments
- [ ] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

**Review by subinasr — APPROVED** [review]

File: app/components/LeftPaneEntries/AssistItem/utils.ts [hunk 1]
```diff
@@ -1,64 +1,81 @@
...
+/*
 type ScaleWidgetAttribute = getType<PartialAttributeType, { widgetType: 'SCALE' }>;
 type SingleSelectWidgetAttribute = getType<PartialAttributeType, { widgetType: 'SELECT' }>;
 type MultiSelectWidgetAttribute = getType<PartialAttributeType, { widgetType: 'MULTISELECT' }>;
 type OrganigramWidgetAttribute = getType<PartialAttributeType, { widgetType: 'ORGANIGRAM' }>;
 type GeoLocationWidgetAttribute = getType<PartialAttributeType, { widgetType: 'GEO' }>;
+*/
+
+export function isValidObject(value: unknown): value is Record<string, unknown> {
+    return typeof value === 'object' && !Array.isArray(value) && value !== null && value !== undefined;
```
Comment (samshara) [inline app/components/LeftPaneEntries/AssistItem/utils.ts]:
```suggestion
    return typeof value === 'object' && !Array.isArray(value) && value !== null;
```

File: app/components/LeftPaneEntries/AssistItem/utils.ts [hunk 2]
```diff
@@ -1,64 +1,81 @@
...
 
-export function createMatrix1dAttr(
-    mappings: Matrix1dMappingsItem[] | undefined,
+export function isValidStringArray(value: unknown): value is string[] {
+    const isArray = typeof value === 'object' && Array.isArray(value) && value !== null && value !== undefined;
+    if (!isArray) {
+        return false;
+    }
+    return value.every((item) => typeof item === 'string');
+}
```
Comment (samshara) [inline app/components/LeftPaneEntries/AssistItem/utils.ts]:
```suggestion
export function isValidStringArray(value: unknown): value is string[] {
    return Array.isArray(value) && value.every((item) => typeof item === 'string');
}

```

File: app/components/LeftPaneEntries/AssistItem/utils.ts [hunk 3]
```diff
@@ -1,64 +1,81 @@
...
+
+// TODO: Write tests
+export function createMatrix1dAttrFromTags(
+    unsafeMappings: Matrix1dValue | undefined,
     widget: Matrix1dWidget,
 ): Matrix1dWidgetAttribute | undefined {
-    if (!mappings || mappings.length <= 0) {
+    // NOTE: We are supposed to get Matrix1dValue, but adding one more validation layer
+    const mappings = unsafeMappings as unknown;
```
Comment (samshara) [inline app/components/LeftPaneEntries/AssistItem/utils.ts:45]: The type of `unsafeMappings` is already defined. Why are we retyping it as `unknown`?

File: app/components/LeftPaneEntries/AssistItem/utils.ts [hunk 4]
```diff
@@ -1,64 +1,81 @@
...
+        validRowKeys,
+        (validRowKey) => validRowKey,
+        (validRowKey) => {
+            const rowDetails = mappings[validRowKey];
+            if (!isValidObject(rowDetails)) {
+                return undefined;
+            }
+            const cellKeys = Object.keys(rowDetails);
+            const validCellKeys = cellKeys.filter((item) => (
+                cellKeysByRowKey?.[validRowKey].includes(item)
```
Comment (samshara) [inline app/components/LeftPaneEntries/AssistItem/utils.ts:70]: Are we sure that `validRowKey` element exists in `cellKeysByRowKey` ?

File: app/components/LeftPaneEntries/AssistItem/utils.ts [hunk 5]
```diff
@@ -70,74 +87,76 @@ export function createMatrix1dAttr(
...
+        validRowKeys,
+        (validRowKey) => validRowKey,
+        (validRowKey) => {
+            const rowDetails = mappings[validRowKey];
+            if (!isValidObject(rowDetails)) {
+                return undefined;
+            }
+            const subRowKeys = Object.keys(rowDetails);
+            const validsubRowKeys = subRowKeys.filter((item) => (
+                subRowKeysByRowKey?.[validRowKey].includes(item)
```
Comment (samshara) [inline app/components/LeftPaneEntries/AssistItem/utils.ts:124]: Are we sure `validRowKey` exists in `subRowKeysByRowKey`?

**Review by subinasr — APPROVED** [review]

**Pull request merged**